editor/src/document/document_message_handler.rs

```diff
@@ -36,6 +36,9 @@ pub enum DocumentsMessage {
 
 #[derive(Debug, Clone)]
 pub struct DocumentsMessageHandler {
+	// Okay, so I think we can switch this over to a hashmap, I think the question here
+	// is what should be the initial capacity. Should it be 1? Do we pay for the cost of
+	// reallocating...?
```

Comment:
Yeah, the options are pretty much hash map or linear search .
We do need to store the order of the documents (well I am actually not sure about that) but that would require some sort of dual data structure

Reply:
Eventually the order of documents won't matter, except as part of the system which handles the workspace tab arrangement.

Reply:
Hey @Keavon @TrueDoctor, I added some more changes based on what @TrueDoctor said with linear search since that seemed like a good idea.

I introduced recycling old IDs, so if you create a new document, we can recycle an ID and then perform a look up to get the index. I'm not sure if this actually introduces more complexity to be honest. Let me know what you two think and if it's not a good idea, I can rework this.